test/unit/test_cusolver_tile/test_potrf.h

```diff
@@ -0,0 +1,138 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <sstream>
+#include "gtest/gtest.h"
+#include "dlaf/blas/enum_output.h"
+#include "dlaf/cuda/error.h"
+#include "dlaf/cusolver/error.h"
+#include "dlaf/lapack/tile.h"
+#include "dlaf/matrix/copy_tile.h"
+#include "dlaf/matrix/tile.h"
+#include "dlaf/memory/memory_view.h"
+#include "dlaf_test/matrix/util_tile.h"
+#include "dlaf_test/matrix/util_tile_blas.h"
+#include "dlaf_test/util_types.h"
+
+namespace dlaf {
+namespace test {
+
+using namespace dlaf::matrix;
+using namespace dlaf::matrix::test;
+using namespace dlaf::tile;
+using namespace testing;
+
+template <class T, bool return_info>
+void testPotrf(const blas::Uplo uplo, const SizeType n, const SizeType extra_lda) {
+  const TileElementSize size_a = TileElementSize(n, n);
+  const SizeType lda = std::max<SizeType>(1, size_a.rows()) + extra_lda;
+
```

Comment:
Note that the CPU version, and the `NonPosDef` version below, of this test has a `SCOPED_TRACE` here. However, I had to remove it here because it turns out `SCOPED_TRACE` is not thread-safe in the sense that it uses a thread local vector internally to push and pop the trace messages. This is problematic if the scoped trace is constructed on one worker thread and destroyed on another, which can happen when yielding the HPX thread (there's at least a `then` in `potrf` that can occasionally yield). In general HPX attempts to reschedule yielded and suspended threads on the same worker thread where they were yielded or suspended, but does not guarantee it with the work-stealing schedulers.

It looks like all other uses of `SCOPED_TRACE` just don't end up yielding while it's in scope.